### DISCUSSION TITLE
System for replacing shuttles is unintuitive/unpredictable

### DISCUSSION THREAD

**Discussion by ExR-dev** [FakeFishGames/Barotrauma#8666] [Ideas and suggestions]
I don't know if the mechanic I am going to discuss is the result of an oversight or an intentional compromise so I apologise if this should have been reported as a bug instead of as feedback. It has caused massive problems for me so I thought it would be worth bringing up in hopes that it can be changed. I'll start out by saying that I happened upon this issue through a pretty specific set of circumstances, but I believe that it could pose a problem even in much more general scenarios. 

The gist of the problem is that replacing a lost drone does not actually create a new instance of said drone, but instead seemingly brings back the lost drone and repairs it's walls. This for example means that items with durability will keep the durability of the lost drone, items that spawned on the drone and later were removed do not get replaced and any circuits on the drone do not get reset and can therefore very easily break (if they depend on a specific initial state, perchance). I realized this after creating a drone that was made to be used and then discarded: it contains items that get used up when the drone is deployed, which renders it useless after the first use as the items cannot be replaced manually. This of course ended up breaking when trying to replace the drone, as the spent items would not get refilled. Considering that the option is called "Replace Lost Shuttles" I would never have expected that a shuttle would keep it's old properties.

As an end-note, this is not relevant to the issue described above but I would also like to suggest making the shuttle replacement-cost based on the actual cost of the shuttle in the editor, as ~900mk will not be a balanced price for every shuttle. The price also doesn't seem to take into account how many shuttles are being replaced, which I also think would be a good addition.

**Comment by cshingle** [comment]
The issue with this is that some subs are more drone than sub, so replacing rather than recovering the drone will cause huge balance issues for them as it will reset the contents. You may be right that it should be called "Recover Lost Shuttles" instead.

**Reply by ExR-dev** [reply]
The problem you've mentioned is one that I would assume to be the case while playing: I would naturally base my decisions around whether I'm ok with losing everything that is on a drone if I were to leave it behind, so I get the impression this is basically a question of which playstyle/difficulty the mechanic should support. With that said I imagine the best option may possibly be a way for the sub creator to choose whether a drone should be replaced or recovered, possibly as a changeable variable on docking hatches/ports or on linked subs?

The way it works now has made multiple sub ideas I have wanted to make unobtainable as far as I can tell, so I do think it would be a beneficial addition in terms of removing creative limitations. Although I do of course understand that it may at the same time be too much work for it to be worth it.
